# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-xcb-private.h
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-xcb-private.h
--- a/gfx/cairo/cairo/src/cairo-xcb-private.h	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-xcb-private.h	Mon May 06 19:25:12 2024 +0000
@@ -226,9 +226,6 @@
     const xcb_query_extension_reply_t *shm;
     xcb_render_sub_pixel_t *subpixel_orders;
 
-    cairo_list_t free_xids;
-    cairo_freepool_t xid_pool;
-
     cairo_mutex_t shm_mutex;
     cairo_list_t shm_pools;
     cairo_list_t shm_pending;
@@ -323,13 +320,6 @@
     return cairo_device_acquire (&connection->device);
 }
 
-cairo_private uint32_t
-_cairo_xcb_connection_get_xid (cairo_xcb_connection_t *connection);
-
-cairo_private void
-_cairo_xcb_connection_put_xid (cairo_xcb_connection_t *connection,
-			       uint32_t xid);
-
 static inline void
 _cairo_xcb_connection_release (cairo_xcb_connection_t *connection)
 {
@@ -453,8 +443,7 @@
 				     cairo_scaled_font_t          *scaled_font,
 				     cairo_glyph_t                *glyphs,
 				     int                           num_glyphs,
-				     cairo_bool_t                  overlap,
-				     cairo_bool_t                  permit_subpixel_antialiasing);
+				     cairo_bool_t                  overlap);
 cairo_private void
 _cairo_xcb_surface_scaled_font_fini (cairo_scaled_font_t *scaled_font);
 
@@ -483,10 +472,6 @@
 				     uint16_t width,
 				     uint16_t height);
 
-cairo_private void
-_cairo_xcb_connection_free_pixmap (cairo_xcb_connection_t *connection,
-				   xcb_pixmap_t pixmap);
-
 cairo_private xcb_gcontext_t
 _cairo_xcb_connection_create_gc (cairo_xcb_connection_t *connection,
 				 xcb_drawable_t drawable,
@@ -494,10 +479,6 @@
 				 uint32_t *values);
 
 cairo_private void
-_cairo_xcb_connection_free_gc (cairo_xcb_connection_t *connection,
-			       xcb_gcontext_t gc);
-
-cairo_private void
 _cairo_xcb_connection_change_gc (cairo_xcb_connection_t *connection,
 				 xcb_gcontext_t gc,
 				 uint32_t value_mask,
@@ -784,16 +765,6 @@
 						      uint32_t                  num_stops,
 						      xcb_render_fixed_t *stops,
 						      xcb_render_color_t *colors);
-#if CAIRO_HAS_XLIB_XCB_FUNCTIONS
-slim_hidden_proto (cairo_xcb_surface_create);
-slim_hidden_proto (cairo_xcb_surface_create_for_bitmap);
-slim_hidden_proto (cairo_xcb_surface_create_with_xrender_format);
-slim_hidden_proto (cairo_xcb_surface_set_size);
-slim_hidden_proto (cairo_xcb_surface_set_drawable);
-slim_hidden_proto (cairo_xcb_device_debug_get_precision);
-slim_hidden_proto_no_warn (cairo_xcb_device_debug_set_precision);
-slim_hidden_proto_no_warn (cairo_xcb_device_debug_cap_xrender_version);
-#endif
 
 cairo_private void
 _cairo_xcb_resources_get (cairo_xcb_screen_t *screen,